backend/TraderX/src/main/java/cmpe451/group6/rest/follow/service/FollowService.java

```diff
@@ -0,0 +1,168 @@
+package cmpe451.group6.rest.follow.service;
+
+import cmpe451.group6.authorization.exception.CustomException;
+import cmpe451.group6.authorization.model.RegistrationStatus;
+import cmpe451.group6.authorization.model.User;
+import cmpe451.group6.authorization.repository.UserRepository;
+import cmpe451.group6.authorization.security.JwtTokenProvider;
+import cmpe451.group6.rest.follow.DTO.FolloweeDTO;
+import cmpe451.group6.rest.follow.model.FollowDAO;
+import cmpe451.group6.rest.follow.model.FollowStatus;
+import cmpe451.group6.rest.follow.repository.FollowRepository;
+import org.modelmapper.ModelMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import javax.servlet.http.HttpServletRequest;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class FollowService {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private JwtTokenProvider jwtTokenProvider;
+
+    @Autowired
+    private FollowRepository followRepository;
+
+    @Autowired
+    private ModelMapper modelMapper;
+
+    /**
+     * 
+     * @param username
+     * @param request
+     * @return info about state
+     */
+    public String followUser(String username, HttpServletRequest request) {
+        User userToFollow = userRepository.findByUsername(username);
+        User currentUser = userRepository
+                .findByUsername(jwtTokenProvider.getUsername(jwtTokenProvider.resolveToken(request)));
+
+        if (userToFollow == null) {
+            throw new CustomException("There is no user named " + username + ".", HttpStatus.NOT_ACCEPTABLE);
+        } else if (userToFollow.getStatus() == RegistrationStatus.PENDING) {
+            throw new CustomException("The user is not activate his/her account. Therefore s/he cannot be followed ",
+                    HttpStatus.PRECONDITION_FAILED);
+        } else if (amIFollowing(username, request)) {
+            throw new CustomException("The user has already been followed", HttpStatus.PRECONDITION_FAILED);
+        }
+
+        FollowDAO temp = new FollowDAO();
+        temp.setFollower(currentUser);
+        temp.setFollowee(userToFollow);
+        temp.setFollowStatus(FollowStatus.PENDING);
+        temp.setId(3); // TODO: this has to be checked. If I don't give id it does not generate
```

Comment:
I think this is not a reliable way of maintaining this. Any idea why this behavior happens?

Reply:
I agree with you @enozcan this is not a reliable way. I think it's because of implementation of followdao model. Maybe, adding "required=false" tag to the Id field resolve the issue.

Reply:
I have found the problem and corrected u can see from latest commit, btw AFAIK required flag is false by default